Micronet memory issue

On current develop, this test:

`tests/models/test_arch_micronet.py`

Fails with an error message:

`FAILED tests/models/test_arch_micronet.py::test_micronet_output - torch.OutOfMemoryError: CUDA out of memory. Tried to allocate 3.94 GiB. GPU 0 has a total capacity of 23.66 GiB of which 67.19 MiB is free. Including non...`

This is on a fairly powerful GPU (24Gb VRAM) so may indicate some underlying issue related to the engine changes, or maybe just a badly set up test. I tested it on two different machines too, so its not a machine-specific problem.

That particular test is set to local-only (as CI machines dont have a GPU) so isnt flagged by the usual CI runs.

--- a/tiatoolbox/models/engine/multi_task_segmentor.py
+++ b/tiatoolbox/models/engine/multi_task_segmentor.py
@@ -3183,7 +3183,11 @@ def _update_tile_based_predictions_array(
             final_combined = previous_predictions_.copy()
             final_combined[keep_mask] = new_predictions_[keep_mask] + max_inst_value
             new_predictions_ = final_combined  # Update for final merge
-            max_inst_value = np.max(new_predictions_[:][:]) + max_inst_value
+            max_inst_value = (
+                np.max(new_predictions_[:][:]) + max_inst_value
+                if np.any(keep_mask)
+                else max_inst_value
+            )
 
         wsi_info_dict[idx]["predictions"][y_start:y_end, x_start:x_end] = (
             new_predictions_

